<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SpriteWidget</class>
 <widget class="QWidget" name="SpriteWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>354</width>
    <height>670</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLineEdit" name="txtPrefixAndName">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="btnRenameSprite">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_4">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Sprite States:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="cmbStates">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QToolButton" name="btnAddState">
       <property name="text">
        <string>...</string>
       </property>
       <property name="arrowType">
        <enum>Qt::NoArrow</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="btnRemoveState">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QToolButton" name="btnRenameState">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QToolButton" name="btnOrderStateBack">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="btnOrderStateForward">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="grpState">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Animation State Info</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QToolButton" name="btnFirstFrame">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="res/DesignerTool.qrc">
              <normaloff>:/icons16x16/media-rewind.png</normaloff>:/icons16x16/media-rewind.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="btnPlay">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="res/DesignerTool.qrc">
              <normaloff>:/icons16x16/media-play.png</normaloff>:/icons16x16/media-play.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="btnLastFrame">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="res/DesignerTool.qrc">
              <normaloff>:/icons16x16/media-forward.png</normaloff>:/icons16x16/media-forward.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0">
           <widget class="QCheckBox" name="chkBounce">
            <property name="text">
             <string>Play Bounce</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="chkReverse">
            <property name="text">
             <string>Play Reverse</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="chkLoop">
            <property name="text">
             <string>Looping</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Update FPS:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz10">
          <property name="text">
           <string>10</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz20">
          <property name="text">
           <string>20</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz30">
          <property name="text">
           <string>30</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz40">
          <property name="text">
           <string>40</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz50">
          <property name="text">
           <string>50</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnHz60">
          <property name="text">
           <string>60</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="SpriteTableView" name="framesView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>335</width>
          <height>0</height>
         </size>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QToolButton" name="btnAddFrames">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnRemoveFrame">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QToolButton" name="btnOrderFrameUp">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnOrderFrameDown">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <action name="actionRenameState">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-rename.png</normaloff>:/icons16x16/generic-rename.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Rename State</string>
   </property>
   <property name="toolTip">
    <string>Rename state</string>
   </property>
  </action>
  <action name="actionAddState">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-add.png</normaloff>:/icons16x16/generic-add.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Add State</string>
   </property>
   <property name="toolTip">
    <string>Add a new state for this sprite</string>
   </property>
  </action>
  <action name="actionRemoveState">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/edit-delete.png</normaloff>:/icons16x16/edit-delete.png</iconset>
   </property>
   <property name="text">
    <string>Remo&amp;ve State</string>
   </property>
   <property name="toolTip">
    <string>Remove the currently selected state</string>
   </property>
  </action>
  <action name="actionOrderStateBackwards">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-left.png</normaloff>:/icons16x16/generic-left.png</iconset>
   </property>
   <property name="text">
    <string>Order State &amp;Backwards</string>
   </property>
   <property name="toolTip">
    <string>Shift the index of the current state back by 1</string>
   </property>
  </action>
  <action name="actionOrderStateForwards">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-right.png</normaloff>:/icons16x16/generic-right.png</iconset>
   </property>
   <property name="text">
    <string>Order State For&amp;wards</string>
   </property>
   <property name="toolTip">
    <string>Shift the index of the current state forward by 1</string>
   </property>
  </action>
  <action name="actionAlignLeft">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignLeft.png</normaloff>:/icons16x16/sprite-alignLeft.png</iconset>
   </property>
   <property name="text">
    <string>Align Left</string>
   </property>
   <property name="toolTip">
    <string>Aligns the sprite's left side to its origin</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Left</string>
   </property>
  </action>
  <action name="actionAlignRight">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignRight.png</normaloff>:/icons16x16/sprite-alignRight.png</iconset>
   </property>
   <property name="text">
    <string>Align Right</string>
   </property>
   <property name="toolTip">
    <string>Aligns the sprite's right side to its origin</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Right</string>
   </property>
  </action>
  <action name="actionAlignUp">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignUp.png</normaloff>:/icons16x16/sprite-alignUp.png</iconset>
   </property>
   <property name="text">
    <string>Align Up</string>
   </property>
   <property name="toolTip">
    <string>Aligns the sprite's top side to its origin</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Up</string>
   </property>
  </action>
  <action name="actionAlignDown">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignBottom.Image.png</normaloff>:/icons16x16/sprite-alignBottom.Image.png</iconset>
   </property>
   <property name="text">
    <string>Align Down</string>
   </property>
   <property name="toolTip">
    <string>Aligns the sprite's bottom side to its origin</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Down</string>
   </property>
  </action>
  <action name="actionAlignCenterVertical">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignVertCenter.png</normaloff>:/icons16x16/sprite-alignVertCenter.png</iconset>
   </property>
   <property name="text">
    <string>Align Center Vertical</string>
   </property>
   <property name="toolTip">
    <string>Centers the sprite vertically to its origin</string>
   </property>
  </action>
  <action name="actionAlignCenterHorizontal">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-alignHorzCenter.png</normaloff>:/icons16x16/sprite-alignHorzCenter.png</iconset>
   </property>
   <property name="text">
    <string>Align Center Horizontal</string>
   </property>
   <property name="toolTip">
    <string>Centers the sprite horizontally to its origin</string>
   </property>
  </action>
  <action name="actionApplyToAll">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/sprite-applyToAll.png</normaloff>:/icons16x16/sprite-applyToAll.png</iconset>
   </property>
   <property name="text">
    <string>Apply to all</string>
   </property>
   <property name="toolTip">
    <string>Applys alignment to all frames in sprite state</string>
   </property>
  </action>
  <action name="actionPlay">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/media-play.png</normaloff>:/icons16x16/media-play.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="toolTip">
    <string>Preview the sprite's animation</string>
   </property>
  </action>
  <action name="actionFirstFrame">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/media-rewind.png</normaloff>:/icons16x16/media-rewind.png</iconset>
   </property>
   <property name="text">
    <string>First Frame</string>
   </property>
   <property name="toolTip">
    <string>Selects the first frame of this sprite's animation</string>
   </property>
  </action>
  <action name="actionLastFrame">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/media-forward.png</normaloff>:/icons16x16/media-forward.png</iconset>
   </property>
   <property name="text">
    <string>Last Frame</string>
   </property>
   <property name="toolTip">
    <string>Selects the last frame of this sprite's animation</string>
   </property>
  </action>
  <action name="actionOrderFrameDownwards">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-down.png</normaloff>:/icons16x16/generic-down.png</iconset>
   </property>
   <property name="text">
    <string>Order Frame &amp;Downwards</string>
   </property>
   <property name="toolTip">
    <string>Shift the selected frame to be played later in the sequence</string>
   </property>
  </action>
  <action name="actionOrderFrameUpwards">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-up.png</normaloff>:/icons16x16/generic-up.png</iconset>
   </property>
   <property name="text">
    <string>Order Frame &amp;Upwards</string>
   </property>
   <property name="toolTip">
    <string>Shift the selected frame to be played earlier in the sequence</string>
   </property>
  </action>
  <action name="actionRemoveFrames">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/edit-delete.png</normaloff>:/icons16x16/edit-delete.png</iconset>
   </property>
   <property name="text">
    <string>Remove Frames</string>
   </property>
   <property name="toolTip">
    <string>Removes the selected frame</string>
   </property>
   <property name="shortcut">
    <string>Del, Backspace</string>
   </property>
  </action>
  <action name="actionImportFrames">
   <property name="icon">
    <iconset resource="res/DesignerTool.qrc">
     <normaloff>:/icons16x16/generic-add.png</normaloff>:/icons16x16/generic-add.png</iconset>
   </property>
   <property name="text">
    <string>Import &amp;Frames</string>
   </property>
   <property name="toolTip">
    <string>Import the selected frames from the Atlas Manager</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SpriteTableView</class>
   <extends>QTableView</extends>
   <header>SpriteModelView.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>txtPrefixAndName</tabstop>
 </tabstops>
 <resources>
  <include location="res/DesignerTool.qrc"/>
 </resources>
 <connections/>
</ui>
